bokeh/models/tools.py

```diff
@@ -371,8 +371,22 @@ class HoverTool(Tool):
     off by setting ``tooltips=None``.
 
     .. warning::
-        Point hit testing is not currently available on all glyphs. Hover tool
-        currently does not work with line or image type glyphs.
+        Hover tool does not currently work with the following glyphs:
+            * annulus
+            * arc
+            * bezier
+            * gear
+            * image
+            * image_rgba
+            * image_url
+            * multi_line
+            * oval
+            * patch
+            * quad
```

Comment:
I opened a PR to fix up the point hit testing code better. I say fix up the example file and remove quad from the list of unsupported glyphs (but continue the discussion of how to make it leas easy to mess up in the original issue #964)